core/sail/fts/lucene/src/test/java/org/eclipse/rdf4j/sail/lucene/AbstractGenericLuceneTest.java

```diff
@@ -0,0 +1,914 @@
+		assertEquals(7, r);
+	}
+
+	@Test
+	public void testQueryWithSpecifiedSubject()
+		throws RepositoryException, MalformedQueryException, QueryEvaluationException
+	{
+		// fire a query with the subject pre-specified
+		TupleQuery query = connection.prepareTupleQuery(QueryLanguage.SERQL, QUERY_STRING);
+		query.setBinding("Subject", SUBJECT_1);
+		query.setBinding("Query", vf.createLiteral("one"));
+		TupleQueryResult result = query.evaluate();
+
+		// check that this subject and only this subject is returned
+		assertTrue(result.hasNext());
+		BindingSet bindings = result.next();
+		assertEquals(SUBJECT_1, (IRI)bindings.getValue("Subject"));
+		assertNotNull(bindings.getValue("Score"));
+		assertFalse(result.hasNext());
+
+		result.close();
+	}
+
+	@Test
+	public void testUnionQuery()
+		throws RepositoryException, MalformedQueryException, QueryEvaluationException
+	{
+		String queryStr = "";
+		queryStr += "PREFIX search: <http://www.openrdf.org/contrib/lucenesail#> ";
+		queryStr += "PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#> ";
+		queryStr += "SELECT DISTINCT ?result { ";
+		queryStr += "{ ?result search:matches ?match1 . ";
+		queryStr += "  ?match1 search:query 'one' ; ";
+		queryStr += "          search:property <urn:predicate1> . }";
+		queryStr += " UNION ";
+		queryStr += "{ ?result search:matches ?match2 . ";
+		queryStr += "  ?match2 search:query 'one' ; ";
+		queryStr += "          search:property <urn:predicate2> . } ";
+		queryStr += "} ";
+
+		// fire a query with the subject pre-specified
+		TupleQuery query = connection.prepareTupleQuery(QueryLanguage.SPARQL, queryStr);
+		TupleQueryResult result = query.evaluate();
+		while (result.hasNext()) {
+			System.out.println(result.next());
+		}
+
+		// check that this subject and only this subject is returned
+		result.close();
+	}
+
+	@Test
+	public void testContextHandling()
+		throws Exception
+	{
+		connection.add(SUBJECT_4, PREDICATE_1, vf.createLiteral("sfourponecone"), CONTEXT_1);
+		connection.add(SUBJECT_4, PREDICATE_2, vf.createLiteral("sfourptwocone"), CONTEXT_1);
+		connection.add(SUBJECT_5, PREDICATE_1, vf.createLiteral("sfiveponecone"), CONTEXT_1);
+		connection.add(SUBJECT_5, PREDICATE_1, vf.createLiteral("sfiveponectwo"), CONTEXT_2);
+		connection.add(SUBJECT_5, PREDICATE_2, vf.createLiteral("sfiveptwoctwo"), CONTEXT_2);
+		connection.commit();
+		// connection.close();
+		// connection = repository.getConnection();
+		// connection.setAutoCommit(false);
+		// test querying
+		assertQueryResult("sfourponecone", PREDICATE_1, SUBJECT_4);
+		assertQueryResult("sfourptwocone", PREDICATE_2, SUBJECT_4);
+		assertQueryResult("sfiveponecone", PREDICATE_1, SUBJECT_5);
+		assertQueryResult("sfiveponectwo", PREDICATE_1, SUBJECT_5);
+		assertQueryResult("sfiveptwoctwo", PREDICATE_2, SUBJECT_5);
+		// blind test to see if this method works:
+		assertNoQueryResult("johannesgrenzfurthner");
+		// remove a context
+		connection.clear(CONTEXT_1);
+		connection.commit();
```

Comment:
It is not necessary. Commit doesnt close the transaction. It just flushes buffer content into repository and clean it afterwards.